controller/controller.go

```diff
@@ -45,24 +61,27 @@ func DefaultConfig() Config {
 type Controller interface {
 	// Submit schedules a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Submit(req Request) (*task.Task, error)
+	Submit(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Start starts a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to launched.
-	Start(req Request) (*task.Task, error)
+	Start(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Stop stops a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Stop(req Request) (*task.Task, error)
+	Stop(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Destroy delets a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to inactive.
-	Destroy(req Request) (*task.Task, error)
+	Destroy(req Request, closer <-chan struct{}) (*task.Task, error)
```

Comment:
We could use the Go "net.Context" which is similar to our requestcontext.Ctx but also provides an API for deadlines/canceling.  https://godoc.org/golang.org/x/net/context#CancelFunc